reindex/reindex.go

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Genome Research Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ ******************************************************************************/
+
+package reindex
+
+import (
+	"log/slog"
+	"os/exec"
+	"time"
+
+	"github.com/wtsi-hgi/go-softpack-builder/config"
+)
+
+// Builder can tell us when a build takes place.
+type Builder interface {
+	SetPostBuildCallback(func())
+}
+
+// Scheduler periodically updates the Spack buildcache index.
+type Scheduler struct {
+	throt   *Throttler
```

Comment:
If you embed the `Throttler` (lookup `go struct embedding`), you can avoid having to forward the Start and Signal methods later on.